posthog/demo/matrix/manager.py

```diff
@@ -58,54 +44,162 @@ def ensure_account_and_run(
-            matrix.simulate()
-        for group_type_index, groups in enumerate(matrix.groups.values()):
+    def run_on_team(self, team: Team, user: User):
+        if not self.pre_save or not self.is_demo_data_pre_saved():
+            if self.matrix.simulation_complete is None:
+                self.matrix.simulate()
+            self.save_analytics_data(self.MASTER_TEAM if self.pre_save else team)
+        if self.pre_save:
+            self.copy_analytics_data_from_master_team(team)
+        self.sync_postgres_with_clickhouse_data(team)
+        self.matrix.set_project_up(team, user)
+        calculate_event_property_usage_for_team(team.pk)
+        team.save()
+
+    def save_analytics_data(self, target_team: Team):
+        if target_team.pk == self.MASTER_TEAM_ID:
+            self.prepare_master_team()
+        bulk_group_type_mappings = []
+        for group_type_index, (group_type, groups) in enumerate(self.matrix.groups.items()):
+            bulk_group_type_mappings.append(
+                GroupTypeMapping(team=target_team, group_type_index=group_type_index, group_type=group_type)
+            )
             for group_key, group in groups.items():
-                save_sim_group(team, cast(Literal[0, 1, 2, 3, 4], group_type_index), group_key, group)
-        sim_persons = matrix.people
-        print(f"[DEMO] Simulated {len(sim_persons)} people in {time.time() - simulation_time:.2f} s")
-        individual_time = time.time()  # FIXME
+                self.save_sim_group(target_team, cast(Literal[0, 1, 2, 3, 4], group_type_index), group_key, group)
+        GroupTypeMapping.objects.bulk_create(bulk_group_type_mappings)
+        sim_persons = self.matrix.people
         for sim_person in sim_persons:
-            sim_person_save_result = save_sim_person(team, sim_person)
-            if sim_person_save_result is not None:  # None is returned if the person wasn't ever seen
-                persons_to_bulk_save.append(sim_person_save_result[0])
-                for distinct_id in sim_person_save_result[1]:
-                    person_distinct_ids_to_bulk_save.append(distinct_id)
-        print(f"[DEMO] Saved (individual part) {len(sim_persons)} people in {time.time() - individual_time:.2f} s")
-        bulk_time = time.time()  # FIXME
-        Person.objects.bulk_create(persons_to_bulk_save)
-        PersonDistinctId.objects.bulk_create(person_distinct_ids_to_bulk_save)
-        print(f"[DEMO] Saved (bulk part) {len(persons_to_bulk_save)} people in {time.time() - bulk_time:.2f} s")
-        matrix.set_project_up(team, user)
-        set_time = time.time()  # FIXME
-        team.save()
-        print(f"[DEMO] Setting project up in {time.time() -set_time:.2f} s")
-        return team
+            self.save_sim_person(target_team, sim_person)
+
+    @classmethod
+    def prepare_master_team(cls):
+        if not Team.objects.filter(id=cls.MASTER_TEAM_ID).exists():
+            organization = Organization.objects.create(name="PostHog")
```

Comment:
sure we want to call it PostHog?

Reply:
I think it makes sense, especially if we make plugins managed in the demo environment like on Cloud, in which case they will this way say "Managed by PostHog"